examples/tcp-echo-server.lua

```diff
@@ -1,32 +1,9 @@
-local uv = require('uv')
-
--- Create listener socket
-local server = uv.new_tcp()
-server:bind('127.0.0.1', 1234)
-
-server:listen(128, function(err)
-  -- Create socket handle for client
-  local client = uv.new_tcp()
-
-  -- Accept incoming connection
-  server:accept(client)
-  print("Client connected")
-
-  -- Relay data back to client
-  client:read_start(function(err, data)
-    -- If error, print and close connection
-    if err then
-      print("Client read error: " .. err)
-      client:close()
-    end
-
-    -- If data is set the client has sent data, if unset the client has disconnected
-    if data then
-      client:write(data)
-    else
-      print("Client disconnected")
-      client:close()
-    end
+local net = require ("net")
 
+local server = net.createServer(function(client)
+  client:on("data", function(data)
```

Comment:
I think the node style has a `:pipe()` method that could make this even easier.

```lua
client:pipe(client)
```

Otherwise we need to also handle shutdown and close properly as is done inside pipe.